### PR TITLE
unixPB: add openjdk devkit installs for Linux on x64, ppc64le, aarch64

### PR DESCRIPTION
**Pull request by sxa**
There are loads of issues with this but as a "starter for 10" it should be functional. Stuff to fix later:
- As a minimum allow for other devkit names in the SHA sum variables since this is hard coded to one per architecture ... Or use GPG signatures since they're being created
- Should we have more families for the install? Suse is currently excluded but we don't have any Suse build systems
- We don't yet have a solution for s390x ...

Related:
- https://github.com/adoptium/temurin-build/issues/3468
- https://github.com/adoptium/temurin-build/issues/3519
- https://github.com/adoptium/temurin-build/pull/3706#issuecomment-2015305336

##### Checklist

- [x] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [ ] playbook changes run through [VPC](https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptium.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access) https://ci.adoptium.net/view/Tooling/job/VagrantPlaybookCheck/1863/
- [ ] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

**Comment by sxa** [comment]
New pushes because the `bxx` number has moved to a more sensible place, and there is now a top level directory in the tarball. Prebiously the filename was defined as:
`devkit_filename: devkit-{{ devkit_version|replace('-b','-{{ ansible_architecture }}-linux-gnu-b') }}.tar.xz`

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    url: https://github.com/adoptium/devkit-binaries/releases/download/{{ devkit_version }}/{{  devkit_filename }}
+    dest: /tmp/devkit.tar.xz
+    force: no
+    mode: 0644
+    checksum: "sha256:{{ lookup('vars', 'csum_' + ansible_architecture) }}"
+  when:
+    - devkit_installed.rc != 0
+  tags: devkit
+
+# Perhaps this should have the `b01` removed ... Or use a symlink?
```
Comment (karianna) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/tasks/main.yml:31]: Comment no longer relevant?

**Review by karianna — CHANGES_REQUESTED** [review]

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml [hunk 2]
```diff
@@ -72,6 +72,12 @@
       tags: [build_tools]
     - role: gcc_11                # JDK19+
       tags: [build_tools]
+    - role: devkit                # JDK19+
+      devkit_version: gcc-11.3.0-Centos7.9.2009-b01
```
Comment (andrew-m-leonard) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml:76]: The devkit needs to be build "b02", as I have been doing lots of testing on b01, and wanted a clean/known build that is essential for reproducible builds. Also aarch64 needs to be Centos7.6 devkit as for reproducible builds by RHEL users, only aarch64 rhel 7.6 is available.
So the versions need to be:
- x64/ppc64le : gcc-11.3.0-Centos7.9.2009-b02
- aarch64 : gcc-11.3.0-Centos7.6.1810-b02

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/vars/main.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
+csum_x86_64: 234df9f9dfad20c329c6a45491e0640c4fe6075d049aca37a19cec886e0efbe9
+csum_aarch64: 2870ab4859cc15b98cbaebf72011789171406d6b54770f95d862fd51e9815ae5
```
Comment (andrew-m-leonard) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/vars/main.yml:3]: Update sha's to match b02 builds

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]

**Review by andrew-m-leonard — APPROVED** [review]
looks good

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/tasks/main.yml (hunk 1 above)
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/tasks/main.yml:31]: No it's still relevant as I reverted a change I did yesterday - {{ devkit_version }} here still has the `bXX` in it so the comment could give people a "TODO" / food for thought

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/vars/main.yml (hunk 3 above)
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/devkit/vars/main.yml]: Done

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml (hunk 2 above)
Comment (sxa) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/main.yml]: Done

**Review by karianna — APPROVED** [review]

**Pull request merged**